src/components/Swap/SwapActionButton/SwapButton.tsx

```diff
@@ -124,7 +121,7 @@ export default function SwapButton({
       <ActionButton
         color={color}
         onClick={onClick}
-        disabled={disabled || (permit2Enabled && permit.state === PermitState.LOADING)}
+        disabled={disabled || (permit2Enabled && allowance.state === AllowanceState.UNKNOWN)}
```

Comment:
how can the allowance state get to unknown?

Reply:
This will occur if tokenAllowance or permitAllowance have not yet been read off of the chain, or if the JSON-RPC's doing so fail. It is essentially a "loading" state.

I am fine re-naming this `LOADING` - please advise.

Reply:
I've renamed this to `LOADING` to clarify.